x-pack/libbeat/autodiscover/providers/aws/elb/provider_test.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elb
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/pkg/errors"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/libbeat/common/bus"
+)
+
+type testEventAccumulator struct {
+	events []bus.Event
+	lock   sync.Mutex
+}
+
+func (tea *testEventAccumulator) add(e bus.Event) {
+	tea.lock.Lock()
+	defer tea.lock.Unlock()
+
+	tea.events = append(tea.events, e)
+}
+
+func (tea *testEventAccumulator) len() int {
+	tea.lock.Lock()
+	defer tea.lock.Unlock()
+
+	return len(tea.events)
+}
+
+func (tea *testEventAccumulator) get() []bus.Event {
+	tea.lock.Lock()
+	defer tea.lock.Unlock()
+
+	res := make([]bus.Event, len(tea.events))
+	copy(res, tea.events)
+	return res
+}
+
+func (tea *testEventAccumulator) waitForNumEvents(t *testing.T, targetLen int, timeout time.Duration) {
+	start := time.Now()
+
+	for time.Now().Sub(start) < timeout {
+		if tea.len() >= targetLen {
+			return
+		}
+		time.Sleep(time.Millisecond)
+	}
+
+	t.Fatalf("Timed out waiting for num events to be %d", targetLen)
+}
+
+func Test_internalBuilder(t *testing.T) {
```

Comment:
```suggestion
func TestInternalBuilder(t *testing.T) {
```

Reply:
This is actually the preferred pattern for naming method tests, inherited from godoc examples where underscore delineates a method vs. a plain function named `InternalBuilder`. See [testing](https://golang.org/pkg/testing/#hdr-Examples) for more info. 

I think the go team should be more clear here IMHO, and I realize that this usage is less common, but I think the underscore makes sense because it's more clear that `internalBuilder` is not exported, vs. capitalizing it.

Reply:
> See testing for more info.

The examples there are for example code, where some convention is actually needed to match certain objects to its examples in docs. There the underscore is reserved for struct methods, that would require a dot otherwise, what wouldn't be valid.
In general I'd prefer to don't use underscores, but I agree that these conventions are not clear enough for these cases, and the point about it being a not exported function is a good one, so as you prefer :slightly_smiling_face: